aviatrix/data_source_aviatrix_network_domains.go

```diff
@@ -0,0 +1,120 @@
+package aviatrix
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/AviatrixSystems/terraform-provider-aviatrix/v2/goaviatrix"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceAviatrixNetworkDomains() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAviatrixNetworkDomainsRead,
```

Comment:
New resources should probably be created with Context, so you should use `ReadWithoutTimeout`. You can look at `data_source_gateway_image` for reference. You will have to update the Read function too.

Reply:
have fixed as the ReadWithoutTimeout and fix the read function, add "ctx context.Contex" as input, also invoke the "GetAllNetworkDomains" add ctx as the input